src/libstore/local-overlay-store.cc

```diff
@@ -176,18 +176,42 @@ void LocalOverlayStore::registerValidPaths(const ValidPathInfos & infos)
 }
 
 
-void LocalOverlayStore::deleteGCPath(const Path & path, uint64_t & bytesFreed)
+void LocalOverlayStore::collectGarbage(const GCOptions & options, GCResults & results)
+{
+    LocalStore::collectGarbage(options, results);
+
+    remountIfNecessary();
+}
+
+
+void LocalOverlayStore::deleteStorePath(const Path & path, uint64_t & bytesFreed)
 {
     auto mergedDir = realStoreDir.get() + "/";
     if (path.substr(0, mergedDir.length()) != mergedDir) {
         warn("local-overlay: unexpected gc path '%s' ", path);
         return;
     }
-    if (pathExists(toUpperPath({path.substr(mergedDir.length())}))) {
-        LocalStore::deleteGCPath(path, bytesFreed);
+
+    StorePath storePath = {path.substr(mergedDir.length())};
+    auto upperPath = toUpperPath(storePath);
+
+    if (pathExists(upperPath)) {
+        std::cerr << "    upper exists" << std::endl;
```

Comment:
```suggestion
        debug("upper exists: %s", path);
```
or something like that?

Reply:
Actually didn't mean to commit this at all! Will fix now.

Reply:
Could be useful for debugging though I suppose :thinking: